File: Server/Game/Bonuses/SpeedBonus.cs
using System;
using System.Diagnostics;
using Server.Game.Entities;

namespace Server.Game.Bonuses
{
    public class SpeedBonus : Bonus
    {
        public SpeedBonus(string bonusType, int lifespan, int bonusAmount) : base(bonusType, lifespan, bonusAmount) { }

        public override void ApplyBonus(Player player)
        {
            Debug.WriteLine("Using speed bonus");
        }

        public override void Update(long delta)
        {
            throw new NotImplementedException();
        }
    }
}

File: Client/src/com/badlogic/core/factory/SpeedBonus.java
package com.badlogic.core.factory;

import com.badlogic.game.GameManager;
import com.badlogic.game.Player;

import java.awt.image.BufferedImage;

public class SpeedBonus extends Bonus {
    public SpeedBonus(GameManager gameManager, BufferedImage sprite) {
        super(gameManager, sprite);
    }

    @Override
    public void applyBonus(Player player) {
        System.out.println("Adding " +  this.bonusAmount + " speed to player.");
    }

    @Override
    public int getBonusAmount() {
        return this.bonusAmount;
    }

    @Override
    public int getLifespan() {
        return this.lifespan;
    }

    @Override
    public void setLifespan(int lifespan) {
        this.lifespan = lifespan;
    }

    @Override
    public void setBonusAmount(int bonusAmount) {
        this.bonusAmount = bonusAmount;
    }

    @Override
    public void update(int delta) {

    }
}
